### ISSUE TITLE
[BUG] Throughput regression on perf run after 4/17.

### ISSUE THREAD

**Issue by mch2** [opensearch-project/OpenSearch#7293]
**Describe the bug**
There has been a throughput regression using http_logs dataset after 4/17. 

![Screen Shot 2023-04-23 at 1 31 44 PM](https://user-images.githubusercontent.com/1594901/234050211-189a795e-45e0-4d80-825b-2b79c9bdc3c9.png)
![Screen Shot 2023-04-23 at 1 31 48 PM](https://user-images.githubusercontent.com/1594901/234050214-2e70c6ae-d2b3-4523-9b2d-15730d31f0f1.png)

This regression was likely introduced with one of [these](https://github.com/opensearch-project/OpenSearch/commits/2.x) commits.

**Comment by mch2** [comment]
This is likely the introduction of enabling segment replication backpressure by default with [this](https://github.com/opensearch-project/OpenSearch/commit/172a78a076e057295912c9a6ff86a286f25faebd#diff-9334f12a7606b43aa9694009031b11ff8330c0a1365c29d37826f1b7a4d4ee8a) commit.

After executing a local run of SO dataset I see a similar regression.
```
ec2-user@ip-172-31-19-240 ~]$ opensearch-benchmark compare --baseline 66647b6c-8cc8-4684-9a20-27c62874e8ba --contender 80dbabdc-c7c1-4f7a-b1e0-698e8a81a3ad
 
   ____                  _____                      __       ____                  __                         __
  / __ \____  ___  ____ / ___/___  ____ ___________/ /_     / __ )___  ____  _____/ /_  ____ ___  ____ ______/ /__
 / / / / __ \/ _ \/ __ \\__ \/ _ \/ __ `/ ___/ ___/ __ \   / __  / _ \/ __ \/ ___/ __ \/ __ `__ \/ __ `/ ___/ //_/
/ /_/ / /_/ /  __/ / / /__/ /  __/ /_/ / /  / /__/ / / /  / /_/ /  __/ / / / /__/ / / / / / / / / /_/ / /  / ,<
\____/ .___/\___/_/ /_/____/\___/\__,_/_/   \___/_/ /_/  /_____/\___/_/ /_/\___/_/ /_/_/ /_/ /_/\__,_/_/  /_/|_|
    /_/
 
 
Comparing baseline
  TestExecution ID: 66647b6c-8cc8-4684-9a20-27c62874e8ba
  TestExecution timestamp: 2023-04-24 01:18:48
  TestProcedure: append-no-conflicts
  ProvisionConfigInstance: external
 
with contender
  TestExecution ID: 80dbabdc-c7c1-4f7a-b1e0-698e8a81a3ad
  TestExecution timestamp: 2023-04-24 00:51:02
  TestProcedure: append-no-conflicts
  ProvisionConfigInstance: external
 
------------------------------------------------------
    _______             __   _____
   / ____(_)___  ____ _/ /  / ___/_________  ________
  / /_  / / __ \/ __ `/ /   \__ \/ ___/ __ \/ ___/ _ \
 / __/ / / / / / /_/ / /   ___/ / /__/ /_/ / /  /  __/
/_/   /_/_/ /_/\__,_/_/   /____/\___/\____/_/   \___/
------------------------------------------------------
 
|                                                        Metric |                     Task |    Baseline |   Contender |     Diff |   Unit |
|--------------------------------------------------------------:|-------------------------:|------------:|------------:|---------:|-------:|
|                    Cumulative indexing time of primary shards |                          |     143.836 |     142.999 | -0.83765 |    min |
|             Min cumulative indexing time across primary shard |                          |     2.32835 |     2.27668 | -0.05167 |    min |
|          Median cumulative indexing time across primary shard |                          |     2.40673 |     2.38605 | -0.02068 |    min |
|             Max cumulative indexing time across primary shard |                          |     2.45847 |      2.5085 |  0.05003 |    min |
|           Cumulative indexing throttle time of primary shards |                          |           0 |           0 |        0 |    min |
|    Min cumulative indexing throttle time across primary shard |                          |           0 |           0 |        0 |    min |
| Median cumulative indexing throttle time across primary shard |                          |           0 |           0 |        0 |    min |
|    Max cumulative indexing throttle time across primary shard |                          |           0 |           0 |        0 |    min |
|                       Cumulative merge time of primary shards |                          |     77.8617 |     75.8006 | -2.06112 |    min |
|                      Cumulative merge count of primary shards |                          |        1327 |        1325 |       -2 |        |
|                Min cumulative merge time across primary shard |                          |     1.08268 |     1.04545 | -0.03723 |    min |
|             Median cumulative merge time across primary shard |                          |     1.28234 |     1.22928 | -0.05307 |    min |
|                Max cumulative merge time across primary shard |                          |     1.64947 |      1.6319 | -0.01757 |    min |
|              Cumulative merge throttle time of primary shards |                          |     10.5335 |      9.5461 | -0.98743 |    min |
|       Min cumulative merge throttle time across primary shard |                          |    0.132583 |    0.125133 | -0.00745 |    min |
|    Median cumulative merge throttle time across primary shard |                          |     0.16245 |    0.153875 | -0.00857 |    min |
|       Max cumulative merge throttle time across primary shard |                          |      0.3117 |      0.3064 |  -0.0053 |    min |
|                     Cumulative refresh time of primary shards |                          |     57.5864 |     57.8795 |   0.2931 |    min |
|                    Cumulative refresh count of primary shards |                          |       11683 |       11651 |      -32 |        |
|              Min cumulative refresh time across primary shard |                          |      0.8724 |    0.852633 | -0.01977 |    min |
|           Median cumulative refresh time across primary shard |                          |    0.921975 |    0.902433 | -0.01954 |    min |
|              Max cumulative refresh time across primary shard |                          |     1.07292 |      1.1418 |  0.06888 |    min |
|                       Cumulative flush time of primary shards |                          |    0.773367 |    0.671017 | -0.10235 |    min |
|                      Cumulative flush count of primary shards |                          |          60 |          60 |        0 |        |
|                Min cumulative flush time across primary shard |                          |  0.00298333 |  0.00451667 |  0.00153 |    min |
|             Median cumulative flush time across primary shard |                          |   0.0112667 |  0.00998333 | -0.00128 |    min |
|                Max cumulative flush time across primary shard |                          |   0.0307667 |   0.0265833 | -0.00418 |    min |
|                                       Total Young Gen GC time |                          |     199.859 |     194.168 |   -5.691 |      s |
|                                      Total Young Gen GC count |                          |       23525 |       23101 |     -424 |        |
|                                         Total Old Gen GC time |                          |           0 |           0 |        0 |      s |
|                                        Total Old Gen GC count |                          |           0 |           0 |        0 |        |
|                                                    Store size |                          |     67.2936 |     67.0412 | -0.25246 |     GB |
|                                                 Translog size |                          | 6.14673e-06 | 6.14673e-06 |        0 |     GB |
|                                        Heap used for segments |                          |           0 |           0 |        0 |     MB |
|                                      Heap used for doc values |                          |           0 |           0 |        0 |     MB |
|                                           Heap used for terms |                          |           0 |           0 |        0 |     MB |
|                                           Heap used for norms |                          |           0 |           0 |        0 |     MB |
|                                          Heap used for points |                          |           0 |           0 |        0 |     MB |
|                                   Heap used for stored fields |                          |           0 |           0 |        0 |     MB |
|                                                 Segment count |                          |        1526 |        1419 |     -107 |        |
|                                                Min Throughput |             index-append |     26145.1 |     26052.6 |  -92.502 | docs/s |
|                                               Mean Throughput |             index-append |     28911.4 |     28165.7 | -745.719 | docs/s |
|                                             Median Throughput |             index-append |     28611.3 |     28027.6 | -583.639 | docs/s |
|                                                Max Throughput |             index-append |     32810.5 |       30220 | -2590.51 | docs/s |
|                                       50th percentile latency |             index-append |     1388.29 |     1389.87 |  1.57956 |     ms |
|                                       90th percentile latency |             index-append |     2148.72 |     2038.49 | -110.224 |     ms |
|                                       99th percentile latency |             index-append |     3188.14 |      3120.6 | -67.5362 |     ms |
|                                     99.9th percentile latency |             index-append |     3518.46 |     3693.98 |  175.521 |     ms |
|                                      100th percentile latency |             index-append |     3827.51 |     4199.15 |  371.642 |     ms |
|                                  50th percentile service time |             index-append |     1388.29 |     1389.87 |  1.57956 |     ms |
|                                  90th percentile service time |             index-append |     2148.72 |     2038.49 | -110.224 |     ms |
|                                  99th percentile service time |             index-append |     3188.14 |      3120.6 | -67.5362 |     ms |
|                                99.9th percentile service time |             index-append |     3518.46 |     3693.98 |  175.521 |     ms |
|                                 100th percentile service time |             index-append |     3827.51 |     4199.15 |  371.642 |     ms |
|                                                    error rate |             index-append |           0 |           0 |        0 |      % |
|                                                Min Throughput | wait-until-merges-finish |    0.102563 |   0.0737063 | -0.02886 |  ops/s |
|                                               Mean Throughput | wait-until-merges-finish |    0.102563 |   0.0737063 | -0.02886 |  ops/s |
|                                             Median Throughput | wait-until-merges-finish |    0.102563 |   0.0737063 | -0.02886 |  ops/s |
|                                                Max Throughput | wait-until-merges-finish |    0.102563 |   0.0737063 | -0.02886 |  ops/s |
|                                      100th percentile latency | wait-until-merges-finish |     9749.68 |     13566.9 |  3817.19 |     ms |
|                                 100th percentile service time | wait-until-merges-finish |     9749.68 |     13566.9 |  3817.19 |     ms |
|                                                    error rate | wait-until-merges-finish |           0 |           0 |        0 |      % |
 
 
-------------------------------
[INFO] SUCCESS (took 0 seconds)
-------------------------------
```

Suggest we revert this change and make this feature opt-in for clusters with indices of mixed replication strategy.  We can make it default when segment replication is enabled at a cluster level.

**Comment by bbarani** [comment]
@Rishikesh1159 Can we use this issue to track and understand the root cause of the perf regression as well?

**Comment by Rishikesh1159** [comment]
@bbarani this issue is reason for performance regression we have seen. @mch2 has looked into this issue of performance regression. We compared benchmark runs locally with this [PR](https://github.com/opensearch-project/OpenSearch/pull/7183) in included and reverting (excluding) the PR. We have seen an improvement after reverting the PR as posted in above [comment](https://github.com/opensearch-project/OpenSearch/issues/7293#issuecomment-1520435862). This [PR](https://github.com/opensearch-project/OpenSearch/pull/7183) (enables backpressure setting by default) is causing the regression in document replication, so we reverted it. We should see the improvement in upcoming nightly benchmark runs
